src/libraries/LPFeeLibrary.sol

```diff
@@ -19,14 +20,33 @@ library LPFeeLibrary {
         return self & DYNAMIC_FEE_FLAG != 0;
     }
 
+    function isValid(uint24 self) internal pure returns (bool) {
+        return self <= MAX_LP_FEE;
+    }
+
     function validate(uint24 self) internal pure {
-        if (self > MAX_LP_FEE) revert FeeTooLarge();
+        if (!self.isValid()) revert FeeTooLarge();
     }
 
     function getInitialLPFee(uint24 self) internal pure returns (uint24 lpFee) {
         // the initial fee for a dynamic fee pool is 0
         if (self.isDynamicFee()) return 0;
-        lpFee = self & STATIC_FEE_MASK;
+        lpFee = self & FEE_MASK;
         lpFee.validate();
     }
+
+    /// @notice converts a fee to an override fee (to be returned by beforeSwap) by setting the top bit of the uint24
+    function asOverrideFee(uint24 self) internal pure returns (uint24) {
```

Comment:
I dont think we should keep this in this library bc core doesnt actually use it

Reply:
its used in our test hook 🤔 

another periphery vs core debate 🤔 

```solidity
    function beforeSwap(address, PoolKey calldata, IPoolManager.SwapParams calldata, bytes calldata)
        external
        override
        returns (bytes4, BeforeSwapDelta, uint24)
    {
        return (IHooks.beforeSwap.selector, BeforeSwapDeltaLibrary.ZERO_DELTA, fee.asOverrideFee());
    }
```